sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -732,8 +713,13 @@ namedExpressionSeq
     : namedExpression (',' namedExpression)*
     ;
 
-transformList
-    : '(' transforms+=transform (',' transforms+=transform)* ')'
+partitionFieldList
+    : '(' fields+=partitionField (',' fields+=partitionField)* ')'
+    ;
+
+partitionField
```

Comment:
It's a bit weird if we allow mixed usage, like `PARTITION BY (year(a), b string)`, shall we just define 2 `PARTITIONED BY` clauses?

Reply:
This is allowed in the grammar to give a better error message later. Otherwise it is confusing that `b string` works sometimes, but not others.

Reply:
I'd understand that we will have assertion to force using only one side of pattern at a time.